# Credit Risk Evaluator

In [1]:
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.model_selection import train_test_split

## Retrieve the Data

The data is located in the Challenge Files Folder:

* `lending_data.csv`

Import the data using Pandas. Display the resulting dataframe to confirm the import was successful.

In [2]:
# Import the data
file_path = Path("Resources/lending_data.csv")
lending_df = pd.read_csv(file_path)
lending_df.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  loan_status  
0                 1       22800            0  
1                 0       13600            0  
2                 0       16100            0  
3                 1       22700            0  
4                 1       23000            0

## Predict Model Performance

You will be creating and comparing two models on this data: a Logistic Regression, and a Random Forests Classifier. Before you create, fit, and score the models, make a prediction as to which model you think will perform better. You do not need to be correct! 

Write down your prediction in the designated cells in your Jupyter Notebook, and provide justification for your educated guess.

*Replace the text in this markdown cell with your predictions, and be sure to provide justification for your guess.*

## Split the Data into Training and Testing Sets

In [4]:
# Split the data into X_train, X_test, y_train, y_test

y = lending_df["loan_status"].values
X = lending_df.drop("loan_status", axis=1)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, stratify=y, random_state=0)
X_train.head()

loan_size  interest_rate  borrower_income  debt_to_income  \
8704     10600.0          7.647            52600        0.429658   
71995    10100.0          7.404            50300        0.403579   
10616     7700.0          6.418            41000        0.268293   
61253    10800.0          7.696            53000        0.433962   
54878     8900.0          6.915            45700        0.343545   

       num_of_accounts  derogatory_marks  total_debt  
8704                 5                 1       22600  
71995                4                 1       20300  
10616                2                 0       11000  
61253                5                 1       23000  
54878                3                 0       15700

In [5]:
print(f'X_train shape: {X_train.shape}')
print(f'X_test shape: {X_test.shape}')
print(f'y_train shape: {y_train.shape}')
print(f'y_test shape: {y_test.shape}')

X_train shape: (38768, 7)
X_test shape: (38768, 7)
y_train shape: (38768,)
y_test shape: (38768,)


## Create, Fit and Compare Models

Create a Logistic Regression model, fit it to the data, and print the model's score. Do the same for a Random Forest Classifier. You may choose any starting hyperparameters you like. 

Which model performed better? How does that compare to your prediction? Write down your results and thoughts in the designated markdown cell.

In [6]:
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression(max_iter=10000)
classifier


LogisticRegression(max_iter=10000)

In [7]:
classifier.fit(X_train, y_train)


LogisticRegression(max_iter=10000)

In [8]:
print(f"Training Data Score: {classifier.score(X_train, y_train)}")
print(f"Testing Data Score: {classifier.score(X_test, y_test)}")

Training Data Score: 0.9926227816756088
Testing Data Score: 0.9917715641766405


In [9]:
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler().fit(X_train)
X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [10]:
from sklearn.ensemble import RandomForestClassifier
clf = RandomForestClassifier(random_state=1, n_estimators=500)

In [11]:
clf.fit(X_train_scaled, y_train)

RandomForestClassifier(n_estimators=500, random_state=1)

In [12]:
print(f'Training Score: {clf.score(X_train_scaled, y_train)}')
print(f'Testing Score: {clf.score(X_test_scaled, y_test)}')

Training Score: 0.9979880313660752
Testing Score: 0.9913588526619893


In [13]:
print(f'Logistic Regression model testing score: {classifier.score(X_test, y_test)}')
print(f'Random Forest Classifier model testing score: {clf.score(X_test_scaled, y_test)}')

Logistic Regression model testing score: 0.9917715641766405
Random Forest Classifier model testing score: 0.9913588526619893


# I would say both train ran sucessful I think the logistic regression model would be best. It has a score of .991 

*Which model performed better? How does that compare to your prediction? Replace the text in this markdown cell with your answers to these questions.*